lib/sassc/functions_handler.rb

```diff
@@ -10,25 +10,47 @@ def setup(native_options)
 
       list = Native.make_function_list(Script.custom_functions.count)
 
-      functs = FunctionWrapper.extend(Script::Functions)
-      functs.options = @options
+      functions = FunctionWrapper.extend(Script::Functions)
+      functions.options = @options
 
       Script.custom_functions.each_with_index do |custom_function, i|
-        @callbacks[custom_function] = FFI::Function.new(:pointer, [:pointer, :pointer]) do |s_args, cookie|
-          length = Native.list_get_length(s_args)
+        @callbacks[custom_function] = FFI::Function.new(:pointer, [:pointer, :pointer]) do |native_argument_list, cookie|
+          native_argument_list_length = Native.list_get_length(native_argument_list)
+          custom_function_arguments = []
 
-          v = Native.list_get_value(s_args, 0)
-          v = Native.string_get_value(v).dup
+          (0...native_argument_list_length).each do |i|
+            native_value = Native.list_get_value(native_argument_list, i)
 
-          s = Script::String.new(Script::String.unquote(v), Script::String.type(v))
+            case Native.value_get_tag(native_value)
+            when :sass_string
+              native_string = Native.string_get_value(native_value)
+              argument = Script::String.new(Script::String.unquote(native_string), Script::String.type(native_string))
 
-          value = functs.send(custom_function, s)
+              custom_function_arguments << argument
```

Comment:
> do we want to add an else here that raises something like NotImplemented for now?

@bolandrm Since hooking into libsass’ own error handling stuff is a little too tricky for me at the moment, I’ve added some code that upon error, spits out a Sass syntax error and also logs to `$stderr`.